src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -286,6 +286,120 @@ def _deploy_arm_template_core(cli_ctx, resource_group_name,
     return sdk_no_wait(no_wait, smc.deployments.create_or_update, resource_group_name, deployment_name, properties)
 
 
+def _remove_comments_from_json(template):
+    from jsmin import jsmin
+
+    minified = jsmin(template)
+    # Get rid of multi-line strings. Note, we are not sending it on the wire rather just extract parameters to prompt for values
+    result = re.sub(r'"[^"]*?\n[^"]*?(?<!\\)"', '"#Azure Cli#"', minified, re.DOTALL)
```

Comment:
`re.DOTALL` changes the behavior of `.`:

https://docs.python.org/3/library/re.html#re.DOTALL

> re.DOTALL
> Make the '.' special character match any character at all, including a newline; without this flag, '.' will match anything except a newline. Corresponds to the inline flag (?s).

But there is no `.` in the regex at all.